File: src/main/java/com/matag/game/turn/phases/beginning/UpkeepPhase.java
package com.matag.game.turn.phases.beginning;

import com.matag.game.status.GameStatus;
import com.matag.game.turn.action._continue.AutocontinueChecker;
import com.matag.game.turn.phases.AbstractPhase;
import com.matag.game.turn.phases.Phase;
import org.springframework.stereotype.Component;

@Component
public class UpkeepPhase extends AbstractPhase {
  public static final String UP = "UP";

  private final DrawPhase drawPhase;

    public UpkeepPhase(AutocontinueChecker autocontinueChecker, DrawPhase drawPhase) {
        super(autocontinueChecker);
        this.drawPhase = drawPhase;
    }

    @Override
  public String getName() {
    return UP;
  }

  @Override
  public Phase getNextPhase(GameStatus gameStatus) {
    return drawPhase;
  }
}

File: src/main/kotlin/com/matag/game/turn/phases/beginning/UpkeepPhase.kt
package com.matag.game.turn.phases.beginning

import com.matag.game.status.GameStatus
import com.matag.game.turn.action._continue.AutocontinueChecker
import com.matag.game.turn.phases.AbstractPhase
import com.matag.game.turn.phases.Phase
import org.springframework.stereotype.Component

@Component
class UpkeepPhase(autocontinueChecker: AutocontinueChecker?, private val drawPhase: DrawPhase) :
    AbstractPhase(autocontinueChecker) {

    override val name = UP

    override fun getNextPhase(gameStatus: GameStatus): Phase {
        return drawPhase
    }

    companion object {
        const val UP: String = "UP"
    }
}
